### PR TITLE
Config improvements

### PR DESCRIPTION
**Pull request by famousgarkin**
fixing some unreasonable defaults and assumptions about configuration I hit when porting to this collection

I’d expect the config role not to do anything out of the box until given variables, but currently the defaults cannot even be overriden to "empty" due to testing for `is defined`

### CODE REVIEW

File: roles/config/tasks/6_domains.yml [hunk 1]
```diff
@@ -5,8 +5,8 @@
     server_url: "{{ satellite_server_url }}"
     name: "{{ item.name }}"
     description: "{{ item.description | default(omit) }}"
-    locations: "{{ item.locations }}"
-    organizations: "{{ item.organizations }}"
-    dns_proxy: "{{ satellite_fqdn | default(omit) }}"
+    locations: "{{ item.locations | default(omit) }}"
+    organizations: "{{ item.organizations | default(omit) }}"
+    dns_proxy: "{{ item.dns_proxy | default(omit) }}"
```
Comment (bherrin3) [inline roles/config/tasks/6_domains.yml:10]: I see that `dns_proxy` can be defined in `satellite_subnets`, but I don't see `dns_proxy` as a key directly defined within `satellite_domains`?

Have a functioning example of this working?

**Review by bherrin3 — COMMENTED** [review]
In general, changes like this, if another team (OASIS) has re-baselined on this as the source of truth, I would think we would want to see feedback and functioning examples of this working as this inherently changes the behavior of the role.

@cojmckee , since you originally submitted this, thoughts?

**Review by bherrin3 — APPROVED** [review]
@famousgarkin I would still like to see an example of the `items` question resolved. 

However, beyond that, I don't have any real concerns. If you need to rebase for your work and are testing / using this, I would approve a merge.

If @cojmckee gets back to us to let us know these are breaking changes, we can make changes at that point in time. I did sync with Greg Hellings and he was not aware of @cojmckee 's current status.

File: roles/config/tasks/6_domains.yml (hunk 1 above)
Comment (famousgarkin) [reply roles/config/tasks/6_domains.yml:10]: TBH I don't know what the `dns_proxy` is for, but it doesn't work for our deploy for multiple reasons:

- it fails without gathering facts, as `satellite_fqdn` defaults to `ansible_fqdn`, applies to some other defaults too, which I commented out
- it fails without the DNS proxy things be configured, whatever that is, but we don't have/need that
- it expects all domains to have same `dns_proxy`, which doesn't seem legit to me

**Comment by famousgarkin** [comment]
closing these as we had to move forward faster, couldn't wait/rely on this collection

**Pull request closed without merging**